src/sdk/namespace_fs.js

```diff
@@ -2085,34 +2085,44 @@ class NamespaceFS {
                 throw new S3Error(S3Error.InvalidObjectStorageClass);
             }
 
-            if (restore_status.state === GlacierBackend.RESTORE_STATUS_CAN_RESTORE) {
+            if (Glacier.is_externally_managed(stat.xattr)) {
+                if (restore_status.state === Glacier.RESTORE_STATUS_RESTORED) {
+                    // If the item is premigrated then its a no-op
+                    // Should result in HTTP: 200 OK
+                    return false;
+                }
+
+                throw new Error('cannot restore externally managed object');
```

Comment:
should we allow restore on externally managed? maybe add an optional config to allow that?